Add flag to disable per test target Github status

purkhusid opened the pull request:
## What?
Adds a new config variable that disables/enables the status-per-test-target funcitonality that exists now.

## Why?
When you have hundreds or thousands of test targets the status indicators become very hard to navigate. In this case it might make more sense for the end user to explore the whole invocation instead.

purkhusid commented:
I haven't had the chance to test this out on our setup because I'm getting cgo related errors when running the container.

I'm building the container on Fedora 32 and I get the following error:
`/app/server/cmd/buildbuddy: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /app/server/cmd/buildbuddy)`

purkhusid commented:
Manage to test this and this seems to work.

tylerwilliams commented on server/build_event_protocol/build_status_reporter/build_status_reporter.go:
```diff
@@ -261,6 +264,16 @@ func (r *BuildStatusReporter) appURL() string {
 
+func (r *BuildStatusReporter) shouldReportStatusPerTest() bool {
+	shouldReportStatusPerTest := r.env.GetConfigurator().GetGithubConfig().StatusPerTestTarget
+
```
Minor nit, but we should ensure that the GithubConfig is non-nil before using it. How about something like?

if githubConfig := r.env.GetConfigurator().GetGithubConfig(); githubConfig != nil {
  if githubConfig.shouldReportStatusPerTest != nil {
    return *githubConfig.shouldReportStatusPerTest
  }
}
return true

tylerwilliams commented on server/build_event_protocol/build_status_reporter/build_status_reporter.go:
```diff
@@ -45,6 +45,7 @@ func NewBuildStatusReporter(env environment.Env, invocationID string) *BuildStat
 }
 
 func (r *BuildStatusReporter) ReportStatusForEvent(ctx context.Context, event *build_event_stream.BuildEvent) {
+	shouldReportStatusPerTest := r.shouldReportStatusPerTest()
```
What do you think about throwing a bool called "reportStatusPerTest" in the BuildStatusReporter struct and initializing it at BuildStatusReporter construction time? That way we're not having to re-reference the config file on every event.

tylerwilliams reviewed (changes requested):
Hey @purkhusid, thanks for the CL, this seems like it will be useful for big projects :) I had two minor nits but otherwise looks good to me.

purkhusid commented:
@tylerwilliams I addressed your comments!
You don't happen to have a solution for the glibc problem I mentioned above? I would love to have this change before you cut your next release but I can't build a functioning docker image.
You don't happen to publish docker images from the master branch?

tylerwilliams commented:
(Pinged you on slack to chat about the GLIBC issue)
